src/js/utils/DSLParserUtil.js

```diff
@@ -0,0 +1,196 @@
+ * The `filters` object is a library with the available filters that can be used
+ * in this context, while the `resultset` is an instance of `List` or `Tree` on
+ * which the filers have to be applied.
+ *
+ * @private
+ * @param {FilterNode} ast - The AST node for the filter operation
+ * @param {Object} filters - An object containing the valid filters that can be used
+ * @param {List} resultset - An instance of List or Tree containing the items to filter
+ *
+ * @returns {function} - Returns the fabricated filter function
+ */
+/* eslint-disable no-unused-vars */
+function filterFunctionFactory(ast) {
+  return function (filters, resultset) {
+    // TODO: Lookup in `filters` object one or more valid filters that can be
+    //       applied on the bound token. We can use the `this.filterType` and
+    //       `this.filterParams` for this purpose. Then apply them on the
+    //       resultset, keeping only the matched results.
+
+    return resultset;
+  };
+};
+/* eslint-enable no-unused-vars */
+
+/**
+ * The following functions are used by the JISON parser in order to parse
+ * the expression into a properly nested set of functions and AST nodes.
+ *
+ * @name DSLParserUtil
+ */
+module.exports = {
+
+  /**
+   * Namespace for the merge operator
+   *
+   * @namesapce
+   */
+  Merge: {
+
+    /**
+     * Combines two filter functions using the AND operator
+     *
+     * @param {Function} f1 - The first operation function
+     * @param {Function} f2 - The second operation function
+     *
+     * @returns {Function} Returns a combined filter function
+     */
+    and(f1, f2) {
+      let ast = new CombinerNode(DSLCombinerTypes.AND, f1.ast, f2.ast);
+
+      return {
+        filter: combineFiltersFactory(ast, f1.filter, f2.filter),
+        ast
+      };
+    },
+
+    /**
+     * Combines two filter functions using the OR operator
+     *
+     * @param {Function} f1 - The first operation function
+     * @param {Function} f2 - The second operation function
+     *
+     * @returns {Function} Returns a combined filter function
+     */
+    or(f1, f2) {
+      let ast = new CombinerNode(DSLCombinerTypes.OR, f1.ast, f2.ast);
+
+      return {
+        filter: combineFiltersFactory(ast, f1.filter, f2.filter),
+        ast
+      };
+    }
+
+  },
+
+  /**
+   * Operator namespace
+   *
+   * @namesapce
+   */
+  Operator: {
+
+    /**
+     * Return filter function for an attribute operator
+     *
+     * @param {String} label - The attribute label
+     * @param {String} text - The attribute value
+     * @param {Number} lstart - The starting position of the label token
+     * @param {Number} lend - The ending position of the label token
+     * @param {Number} vstart - The starting position of the value token
+     * @param {Number} vend - The ending position of the value token
+     *
+     * @returns {Function} Returns a filter function
+     */
+    attrib(label, text, lstart, lend, vstart, vend) {
```

Comment:
Nope, that's not a JISON requirement, and yes, `attribute` is a better name. Fixed.